The Bluetooth Low Energy GAP and Connection Management serialization (#4161)
nrfconnect/sdk-nrf: samples/bluetooth/rpc_host/README.rst

.. _ble_rpc_host:

Bluetooth: Host for nRF RPC BLE

> ###############################

Reviewer: Just for the consistency, leave a blank line after a heading.

The nRF RPC Host sample demonstrates how to run the Bluetooth Low Energy stack on the another device or CPU using :ref:`nrfxlib:nrf_rpc` that exchange the function calls between cores using `Remote Procedure Calls (RPC)`_.
The sample makes use of the Bluetooth Low Energy on the Application core of an nRF5340 DK when full stack is running on the Network Core.

Overview
********

.. note::

>    Currently a serialization of the :ref:`zephyr:bt_gap` and :ref:`zephyr:bluetooth_connection_mgmt` is supported.

Reviewer: ```suggestion
   Currently, serialization of the :ref:`zephyr:bt_gap` and :ref:`zephyr:bluetooth_connection_mgmt` is supported.
```

> The host (Network Core) is running full Bluetooth Low Energy stack.

Reviewer: network core

The application receives serialized function calls which are decoded and executed and next it sends a response data to the client (Application core).

When the sample starts, it displays the welcome prompt.

Requirements
************

The sample supports the following development kits:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf5340dk_nrf5340_cpunet

Building and Running
********************
.. |sample path| replace:: :file:`samples/bluetooth/rpc_host`

.. include:: /includes/build_and_run.txt

.. _rpc_host_testing:

Testing
=======

This sample must be build with the same Bluetooth configuration as the Application core sample.
For more details see: :ref:`ble_rpc`.
On the Application Core you can build the :ref:`zephyr: bluetooth-beacon-sample`.

> It works out of the box with this sample and does not require configuration change.

Reviewer: Rather: out of the box

> In the Beacon sample directory you can invoke:

Reviewer: Also, leave a blank line before the .. code-block

   .. code-block:: console
      west build -b nrf5340dk_nrf5340_cpuapp -- -DBT_RPC=y

> 

Reviewer: You have one extra blank line here.

After programming the sample to your development kit, test it by performing the following steps:

1. Connect the dual core development kit to the computer using a USB cable.
   The development kit is assigned a COM port (Windows) or ttyACM device (Linux), which is visible in the Device Manager.
#. |connect_terminal|
#. Reset the development kit.

> #. Observe that the terminal connected to the Network core display: "Starting nRF PRC bluetooth host"

Reviewer: ```suggestion
#. Observe that the terminal connected to the network core display: "Starting nRF PRC bluetooth host"
```

> #. On the terminal connected to the Application core you can observe your Bluetooth application is running.

Reviewer: ```suggestion
#. On the terminal connected to the application core you can observe your Bluetooth application is running.
```

.. _ble_rpc_host:

Bluetooth: Host for nRF RPC Bluetooth Low Energy
################################################

> The nRF RPC Host sample demonstrates the Bluetooth Low Energy (LE) stack on another device or CPU using :ref:`nrfxlib:nrf_rpc` that exchanges function calls between cores using `Remote Procedure Calls (RPC)`_.

Author: Done

> The sample uses Bluetooth LE on the application core of an nRF5340 DK when the full stack is running on the network core.

Author: Done

> The nRF RPC Host sample demonstrates the Bluetooth Low Energy (LE) stack with the :ref:`nrfxlib:nrf_rpc` library which exposes the stack's interface to another device or CPU using `Remote Procedure Calls (RPC)`_.
> On the nRF53 device, this sample is supposed to run on the network core and it provides the Bluetooth LE functionality for the application core.

Reviewer: ```suggestion
The nRF RPC Host sample demonstrates the Bluetooth Low Energy (LE) stack with the :ref:`nrfxlib:nrf_rpc` library, which exposes the stack's interface to another device or CPU using `Remote Procedure Calls (RPC)`_.
On an nRF53 device, this sample is supposed to run on the network core and it provides the Bluetooth LE functionality for the application core.
```

Overview
********

.. note::
   Currently, serialization of the :ref:`zephyr:bt_gap` and :ref:`zephyr:bluetooth_connection_mgmt` is supported.

> The host (Network Core) is running full Bluetooth LE stack.

Author: Done

> The host (network core) is running full Bluetooth LE stack.

Reviewer: ```suggestion
The host (network core) is running the full Bluetooth LE stack.
```

> The application receives serialized function calls that it decodes and executes, then sends response data to the client (application core).

Author: Done

> When the sample starts, it displays the welcome prompt "Starting nRF PRC bluetooth host".

Author: Fixed

Requirements
************

The sample supports the following development kits:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf5340dk_nrf5340_cpunet

Building and running
********************

.. |sample path| replace:: :file:`samples/bluetooth/rpc_host`

.. include:: /includes/build_and_run.txt

.. _rpc_host_debug:

Debug build
===========

> You can build the sample with a debugging configuration.

Reviewer: ```suggestion
To build the sample with a debugging configuration, use the following command:
```

.. code-block:: console

   west build samples/bluetooth/rpc_host -- -DCONFIG_OVERLAY=overlay-debugging.conf

.. _rpc_host_testing:

Testing
=======

Build the sample with the same Bluetooth configuration as the application core sample.
For more details, see: :ref:`ble_rpc`.
Build the :ref:`zephyr:bluetooth-beacon-sample` on the application core.
This sample works out of the box and does not require configuration changes.
In the Beacon sample directory, invoke:

   .. code-block:: console

      west build -b nrf5340dk_nrf5340_cpuapp -- -DCONFIG_BT_RPC=y

After programming the sample for your development kit, test it by performing the following steps:

1. Connect the dual core development kit to the computer using a USB cable.
   The development kit is assigned a COM port (Windows) or ttyACM device (Linux), which is visible in the Device Manager.
#. |connect_terminal|
#. Reset the development kit.
#. Observe that the terminal connected to the network core display: "Starting nRF PRC bluetooth host"

> #. On the terminal connected to the application core you can observe your Bluetooth application is running.

Reviewer: ```suggestion
#. On the terminal connected to the application core, you can observe your Bluetooth application is running.
```
